core/app/models/spree/product.rb

```diff
@@ -103,6 +103,11 @@ def find_or_build_master
     self.whitelisted_ransackable_associations = %w[stores variants_including_master master variants]
     self.whitelisted_ransackable_attributes = %w[slug]
 
+    # This enables "Show Deleted" checkbox feature in admin/products page
```

Comment:
I'm not sure about this comment line. If we'll add other scopes later this line had to be removed/changed in something more generic. I'm for removing it since we can understand why this code is here by using git history.